types/es-abstract/es2015.d.ts

```diff
@@ -0,0 +1,160 @@
+		value?: TNext,
+	): IteratorResult<T, TReturn>;
+	IteratorComplete(iterResult: IteratorResult<unknown>): boolean;
+	IteratorValue<T = never, TReturn = never>(iterResult: IteratorResult<T, TReturn>): T | TReturn;
+	IteratorStep<T>(iterator: Iterator<T>): IteratorYieldResult<T> | false;
+	IteratorClose<T>(iterator: Iterator<unknown, unknown, unknown>, completion: () => T): T;
+	CreateIterResultObject<T>(value: T, done: boolean): IteratorResult<T, T>;
+
+	RegExpExec(R: RegExp | { exec(string: string): RegExpExecArray | null }, S: string): RegExpExecArray | null;
+	ArraySpeciesCreate<T>(originalArray: T[], length: number): T[];
+
+	CreateDataProperty(O: object, P: ESPropertyKey, V: unknown): boolean;
+	CreateDataPropertyOrThrow(O: object, P: ESPropertyKey, V: unknown): boolean;
+
+	ObjectCreate(proto: object | null, internalSlotsList?: []): unknown;
+	AdvanceStringIndex(S: string, index: number, unicode: boolean): number;
+
+	CreateMethodProperty(O: object, P: ESPropertyKey, V: unknown): boolean;
+	DefinePropertyOrThrow(O: object, P: ESPropertyKey, desc: ES5.PropertyDescriptor): boolean;
+	DeletePropertyOrThrow(O: object, P: ESPropertyKey): boolean;
+
+	readonly EnumerableOwnNames: typeof Object.keys;
+
+	// tslint:disable-next-line: ban-types
+	thisNumberValue(value: number | Number): number;
+
+	// tslint:disable-next-line: ban-types
+	thisBooleanValue(value: boolean | Boolean): boolean;
+
+	// tslint:disable-next-line: ban-types
+	thisStringValue(value: string | String): string;
+	thisTimeValue(value: Date): number;
+
+	SetIntegrityLevel(O: object, level: 'sealed' | 'frozen'): boolean;
+	TestIntegrityLevel(O: object, level: 'sealed' | 'frozen'): boolean;
+
+	OrdinaryHasInstance(C: unknown, O: object): boolean;
+	OrdinaryHasProperty(O: object, P: ESPropertyKey): boolean;
+
+	// tslint:disable-next-line: ban-types
+	InstanceofOperator(O: object, C: Function | { [Symbol.hasInstance](O: unknown): boolean }): boolean;
+
+	IsPromise(x: unknown): x is Promise<unknown>;
+	ValidateAndApplyPropertyDescriptor(
+		O: undefined,
+		P: unknown,
+		extensible: boolean,
+		Desc: ES5.PropertyDescriptor,
+		current?: ES5.PropertyDescriptor,
+	): boolean;
+	ValidateAndApplyPropertyDescriptor(
+		O: object | undefined,
+		P: ESPropertyKey,
+		extensible: boolean,
+		Desc: ES5.PropertyDescriptor,
+		current?: ES5.PropertyDescriptor,
+	): boolean;
+	OrdinaryDefineOwnProperty(O: object, P: ESPropertyKey, Desc: ES5.PropertyDescriptor): boolean;
+	OrdinaryGetOwnProperty(O: object, P: ESPropertyKey): ES5.PropertyDescriptor | undefined;
+
+	ArrayCreate(length: number, proto?: object | null): unknown[];
+	ArraySetLength(A: unknown[], Desc: ES5.PropertyDescriptor): boolean;
+	CreateHTML(string: unknown, tag: string, attribute: string, value?: unknown): string;
+
+	GetOwnPropertyKeys(O: object, Type: 'String'): string[];
+	GetOwnPropertyKeys(O: object, Type: 'Symbol'): symbol[];
+	GetOwnPropertyKeys(O: object, Type: 'String' | 'Symbol'): string[] | symbol[];
```

Comment:
this isn't a possible return type for the function.

Reply:
This is because of https://github.com/microsoft/TypeScript/issues/14107.